packages/scene-composer/src/components/WebGLCanvasManager.tsx

```diff
@@ -95,6 +71,15 @@ export const WebGLCanvasManager: React.FC = () => {
     }
   };
 
+  const esc = useCallback(() => {
+    window.addEventListener('keyup', (e: KeyboardEvent) => {
+      if (e.key === 'Escape' && !!addingWidget) {
+        setAddingWidget(undefined);
+      }
+    });
+    return window?.removeEventListener('keyup', setAddingWidget as any);
+  }, [addingWidget]) 
```

Comment:
I'm not sure I agree with adding key handling to the window, but this is not blocking since we're only doing it while the editing state is "active".

Keep in mind, in both the appkit and grafana case, we do not own the page, and attaching anything to window risks polluting the global namespace of the host application. We should instead add this key handler to like the canvas or the overall viewer node.

Reply:
I've been trying to figure out how to get it off the `window`, but haven't had luck. Will continue exploring this, but not promising that it will be fixed in the upcoming commit.

Reply:
is it possible to have the canvas listen for the keypress? you can get it from the R3F state 
```
const { gl } = useThree()
gl.domElement.addEventListener('keypress', () => {});
```